### ISSUE TITLE
Fish Audio S2 Pro is out!

### ISSUE THREAD

**Issue by Mixomo** [FranckyB/Voice-Clone-Studio#77]
Hello! I just found out that Fish Audio S2 has been released for the general public! 
The tests I was able to run online surpassed the quality of the quick cloning features of Qwen TTS and Vibevoice! 

Could you please add it to Voice Clone Studio? 

Thanks in advance!

https://fish.audio/es/blog/fish-audio-open-sources-s2/

**Comment by Mixomo** [comment]
> <img width="928" height="678" alt="Image" src="https://github.com/user-attachments/assets/a2b44f53-ccbb-4218-a7a0-a937ac03046f" /> @Mixomo 
> I made my s2.cpp GUI it works 👍simple and it does the Job quite well. VibeCoded but it works 😁
> 
> [out.wav](https://github.com/user-attachments/files/26224823/out.wav)
> 
>  

Excellent! Great job! What better way than to tackle the problem at its root and build something from scratch focused only on what you need. 😁

Last night I made corrections based on your reports in my repository. If you'd like, you can open CMD in the cloned repo folder and make a `git pull`to update the repo with the corrections. (Or cloning from scratch without deleting the venv folder and sample folder) 

But I'm glad you have your CCP version working! 👍

**Comment by cynical2010** [comment]
@FranckyB — Yeah — it’s much better than the new Voxtral I tested recently. It’s newer than Fish, comes with 9 languages, but doesn’t have voice cloning. It’s also quite good, but Fish has more emotion; I read about 15,000 tags or something like that.

So a big yes from my side as well, as @Mixomo already advertised. 😁

<img width="859" height="296" alt="Image" src="https://github.com/user-attachments/assets/5bd4a0eb-ca79-4dd0-9763-62ba9ac05b5c" />

It’s a bit on the slower side, but the quality makes it worth it.

Quality I would rank it above Qwen or Vibe.
For speed, it’s almost like Vibe on my GPU.

Qwen is very quick; the new Voxtral is insanely fast maybe faster then the quick qwen mode, but the lack of voice cloning is a bummer.

**Comment by cynical2010** [comment]
nice to see @FranckyB is back from his trip to hawaii 😂 welcome back 👍

**Comment by Mixomo** [comment]
hey @cynical2010 Voxtral support voice cloning with reference audio! 

The audio quality itself may not be as good as Fish Speech, but it offers a certain level of stability and predictability in its results, which makes it very appealing for scenarios involving very long texts. 

<img width="2110" height="1200" alt="Image" src="https://github.com/user-attachments/assets/dd35365c-e1c5-48bd-9833-119905f2cb00" />

https://huggingface.co/spaces/mistralai/voxtral-tts-demo

**Comment by cynical2010** [comment]
> hey [@cynical2010](https://github.com/cynical2010) Voxtral support voice cloning with reference audio!
> 
> The audio quality itself may not be as good as Fish Speech, but it offers a certain level of stability and predictability in its results, which makes it very appealing for scenarios involving very long texts.
> 
> <img alt="Image" width="2000" height="1200" src="https://private-user-images.githubusercontent.com/108157310/570924470-dd35365c-e1c5-48bd-9833-119905f2cb00.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzQ3NTc0NzYsIm5iZiI6MTc3NDc1NzE3NiwicGF0aCI6Ii8xMDgxNTczMTAvNTcwOTI0NDcwLWRkMzUzNjVjLWUxYzUtNDhiZC05ODMzLTExOTkwNWYyY2IwMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMyOVQwNDA2MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MzE2N2NkZGQxZmIwYTRjNGQ1MjllNjE3NGNlMjJjYWQ3MDYxZTVhOTNiNWM5YWQ5MDdhZjRhNmNkMDNlMGVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.cbVJu90dKoVXoZwm_G60_NKCjjhgOqJWDh2ZH6oAz9Q">
> https://huggingface.co/spaces/mistralai/voxtral-tts-demo
> 
> EDIT:
> 
> Ups, voice clone its only available at Mistral [AI Studio ](https://console.mistral.ai/build/audio/text-to-speech)
> 
> <img alt="Image" width="1199" height="151" src="https://private-user-images.githubusercontent.com/108157310/570924899-f88951ca-f39a-48f4-9d3a-4a30f9949bcb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzQ3NTc0NzYsIm5iZiI6MTc3NDc1NzE3NiwicGF0aCI6Ii8xMDgxNTczMTAvNTcwOTI0ODk5LWY4ODk1MWNhLWYzOWEtNDhmNC05ZDNhLTRhMzBmOTk0OWJjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMyOVQwNDA2MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNzg2N2Q2ODA4MTU2OTYzNDAyZjQxNjRiOTg1ZTg5NTI5ZTdmM2M3ZjA4YjI3OWU0MmU2NGQ2NDhhOWJkMWYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4t9Ih6LpskXbRMAwp21Y3bnvZDkr4eeUueS32TA0W4c">

yeah only the paid api 😣

done some short testing on your new provided s2 gui seem's to do the work no error's yet 👍😉 @Mixomo

**Comment by cynical2010** [comment]
> hey [@cynical2010](https://github.com/cynical2010) I want to thank you so much for suggesting Ninja: it was the missing piece I needed to achieve speeds comparable to Linux on the PyTorch engine.
> 
> As for the CPP engine, I also managed to optimize its CPU usage a bit better to squeeze out a little more speed.
> 
> But honestly, where the speed really shines is in the PyTorch engine—though, of course, you need at least 24 GB of VRAM.
> 
> As for compilation speed within the installation batch file, it now uses Ninja to orchestrate all the paths and dependencies, but it still has to use the Visual Studio compiler, so there isn’t a noticeable speed improvement in that regard.
> 
> one more time, thank you 😃

Yes it works with msvc together and makes it a bit faster in some instances. And to see you get some positive results by using it makes it even better 😉

**Comment by cynical2010** [comment]
@FranckyB I'm switched over to Linux and your Voice Clone Studio works like a charm 🔥👍️  
With all those juicy things like FlashAttention and fast Qwen, it's even faster than in Microslop 🤣

Any news on implementing Fishy McFish (Fish S2)?

**Comment by FranckyB** [comment]
I've pushed a version to Dev, it doesn't support GGUF presently or Lora Training.
I've also added an LLM preset that uses tags.  At the same time, I've added a filtering of tags for other models.  By this I mean, if you use text that had tags for Fish, they'll get removed automatically when using the other models.

I don't want to have to resort to forcing the user to install Visual Studio 2022 to compile stuff for GGUF, so it's either I don't support it, or simply include the compiled version with the app.  Though would be impossible for me to do for mac.

Unfortunately I'm noticing a big issue.  When trying to generate text from long prompts... the volume just gets lower and lower?!
<img width="1088" height="74" alt="Image" src="https://github.com/user-attachments/assets/04df3028-560d-4767-8c6e-59d8e365ce92" />

I can't seem to find the cause...   so this makes it a bit unusable

And it's so slow, even on a 5090?!  🤦

Mixomo, have you encountered this issue?

**Comment by Mixomo** [comment]
@FranckyB 
The low volume issue only happened to me once with a single reference audio file. (And now that I remember, it was in the gguf version and at the lowest quantization; I was testing the degradation of the most compressed model).
This phenomenon may vary depending on the reference audio or its duration. It might be recommend to enable Split by paragraph for longer texts. 
However, as I said, I haven't experienced the low volume problem again and I can do long texts in a single pass without any issues. 

Regarding speed: compilation is a must. But let me try your dev branch cleanly; perhaps I can add my pre-compiled cache folder from my S2 Pro repository. Perhaps with that folder and the right arguments we don't force the user to install extra things or to compile it themselves. 

I'll let you know as soon as I finish the tests.

**Comment by FranckyB** [comment]
Thanks Mixomo,

Any help would be appreciated.  

I've just pushed a small fix, so before doing a pull request, perhaps merge in the small fix.

I've noticed that in your version, your default parameters seem to differ from the original.  
Am I mistaken or was there a reason?  (It did seem like yours got me better results)

**Comment by Mixomo** [comment]
@FranckyB There's no particular reason. I think I remember them being the original values from the Fish Speech repository. If I changed them at some point, I don't recall, given the number of experiments I did. 😅

**Comment by cynical2010** [comment]
as from my test's i did not get any kinda lowering of the Volume as you mentioned @FranckyB 

so far the highest Q8 is a must from my test's i made with S2 lower than Q8 degrades to much for my taste.
Without an GGUF usage it's pretty much unusable to me anyway.

@Mixomo i tried the Linux version from you and the script was kinda broken could not get it functioning on my side i will give it another try the next days but it seems to dropped out when the compile comes in.

**Comment by cynical2010** [comment]
Rewrote a part of the install.sh to get it run for me if needed I can send it out to you 

I had to make the compiler work without vulkan and such based on cuda 12.8 

@Mixomo 

(edit) sending would be stupid I saw I had to fix a lot more the gradio went nuts too but it seem's too be working now 🤣👍️

**Comment by cynical2010** [comment]
[McFish.wav](https://github.com/user-attachments/files/26477177/McFish.wav)

it works 👍️

**Comment by Mixomo** [comment]
@FranckyB 

Hey, I managed to get Fish Speech running at full speed on Windows without requiring users to install C++ or Visual Studio. 
The only things that need to be installed for the compilation task (which is done only once and then saved to disk) are Triton-windows and Ninja (along with a few other missing libraries required for Fish Speech to work on new installations). 

On my 3090, I went from 3–6 s/it to 18–24 s/it 

Everything is in this gist:

https://gist.github.com/Mixomo/0ebad3643137a3298b0b7b15a55c4b9f

(all of this only installed, modified and tested under Windows)

and a breakdown:

### Optimization Summary (script-by-script)

#### `voice_clone_studio.py` (Main Entry)
*   **Inductor Persistence**: Configured `TORCHINDUCTOR_FX_GRAPH_CACHE` and persistent cache directories to ensure optimized kernels are saved to `models/.cache/` and reused across app restarts.
*   **Runtime Auto-Patcher**: Added `auto_patch_inductor()`, a safety function that silently scans the `venv` for the known `cluster_dims` AttributeError in PyTorch Inductor (common on Windows dev builds) and patches it automatically at startup.
*   **Performance Tweak**: Enabled `force_bit32_indexing` to reduce indexing overhead and avoid Windows-specific 64-bit integer latency in Triton kernels.
*   **Status Logs**: Added English console notifications using the `[FISH SPEECH]` prefix to report the state of the persistent kernel cache.

#### `modules/core_components/ai_models/tts_manager.py` (Engine Logic)
*   **Dynamic Token Calculation**: Replaced fixed/zero token generation with a smart heuristic: `(Text Length * 1.5) + 100 tokens padding`. This prevents VRAM leaks and ensures complete sentence generation.
*   **Cache UI Reporting**: Enabled real-time feedback in the Gradio progress bar (`[FISH SPEECH] Using cached kernels...`) so you can distinguish between fast cached runs and slow compilation runs.
*   **Engine Handlers**: Synchronized parameters like `split_by_paragraph` to ensure the core engine respects UI settings.

#### `modules/core_components/tools/voice_clone.py` (UI & Tool Logic)
*   **Split-Paragraph Silence**: Implemented a **0.5-second silence buffer** (`np.zeros`) between clips during split-by-paragraph generation. This works for all engines (Qwen3, VibeVoice, Fish, etc.) to ensure a natural flow in the combined preview.
*   **Handler Synchronization**: Updated the `generate_audio_handler` signature and the `Generate` button wiring to properly pass the `split_paragraph` flag, ensuring the silence logic is actually triggered.

#### `requirements.txt` (Environment)
*   **Performance Stack**: Added `ninja` and `triton-windows` to ensure anyone installing the requirements has the necessary tools for Inductor compilation.
*   **Codec Fixes**: Added explicitly missing sub-dependencies for the Fish Speech DAC codec (`flatten-dict`, `tabulate`, `randomname`, `markdown2`, etc.) to resolve previous `ModuleNotFoundError` crashes on clean installs.

#### `config.json` (Defaults)
*   **Stability**: Synchronized the default values for the Fish Speech engine to match the newly implemented performance profiles.

**Comment by FranckyB** [comment]
Hi Mixomo, 

Any chance you could do a pull request of your changes?

But I'd change using the the config file for anything important though.  It was always meant to be disposable. 
As you can delete it to reset everything back to default.  Anything that needs to set should be done in constant.py.

When starting the app, the constant file is recreated if it doesn't exist, using values from constant.

**Comment by FranckyB** [comment]
Thanks Mixomo! 🎉

It's merged into Dev!
I wont' have time to test it today unfortunately, but it's up in dev for anyone to test.

**Comment by Mixomo** [comment]
@FranckyB I just did a clean reinstall of your dev branch, and everything is working fine. I just forgot to add `tensorboard `to requirements.txt, which is why Fish Speech was throwing an error at first. So all you need to do is add `tensorboard `to your `requirements.txt` in the dev branch. It doesn't make sense to open a pull request for a single line of code haha.

**Comment by FranckyB** [comment]
@Mixomo, I've pushed the fix with tensorboard. 

But I'm having issues on my end.
The first time run is brutal, as it takes a very long time.  
We`ll have to add a message in console so the user knows.  I know there is one at launch, but might be good to let them know when we generate it.

But also, it then crashed and never completes.  I'm looking into it now.

**Comment by Mixomo** [comment]
Ok, let me know your findings. Since I didn't have any problems with a clean install, I have no way of debugging what might be causing your problem. Perhaps our dear Claude will give us some ideas 😅

**Comment by FranckyB** [comment]
Ok, I figured it out.

I was using a newer version of Pytorch, so I added support for it in auto_patch_inductor().
Hot Damn it's fast now, haha!  Awesome work Mixomo!

I've also added a message just before it's starts caching so the user knows what's happening.
Running a few test and then releasing it.

**Comment by Mixomo** [comment]
Awesome! Glad to hear that! 🔥🔥🔥

**Comment by FranckyB** [comment]
Ok, I've pushed it to Dev.
Did a couple of small tweaks and updated the update.md file.

Thanks again Mixomo!

**Comment by Mixomo** [comment]
My pleasure!

**Comment by cynical2010** [comment]
so i'll stay with the GGUF version the other one is too much for my sys to handle so thx for the GGUF version and the others have fun with the big Fishy 😉
@Mixomo
